test_get_exif_tags and test_exif_gps fail with sigal 2.1 and pillow >= 7

When rebuilding sigal for tests against python-pillow >= 7.0.0 I'm running into the following issues:

```
=================================== FAILURES ===================================
______________________________ test_get_exif_tags ______________________________

    def test_get_exif_tags():
        test_image = '11.jpg'
        src_file = os.path.join(CURRENT_DIR, 'sample', 'pictures', 'dir1', 'test1',
                                test_image)
        data = get_exif_data(src_file)
        simple = get_exif_tags(data, datetime_format='%d/%m/%Y')
>       assert simple['fstop'] == 3.9
E       KeyError: 'fstop'

tests/test_image.py:148: KeyError
----------------------------- Captured stderr call -----------------------------
Unknown format for ExposureTime: 0.00100603
------------------------------ Captured log call -------------------------------
INFO     sigal.image:image.py:350 Unknown format for ExposureTime: 0.00100603
________________________________ test_exif_gps _________________________________

tmpdir = local('/tmp/pytest-of-builduser/pytest-0/test_exif_gps0')

    def test_exif_gps(tmpdir):
        """Test reading out correct geo tags"""

        test_image = 'flickr_jerquiaga_2394751088_cc-by-nc.jpg'
        src_file = os.path.join(CURRENT_DIR, 'sample', 'pictures', 'dir1', 'test1',
                                test_image)
        dst_file = str(tmpdir.join(test_image))

        settings = create_settings(img_size=(400, 300), copy_exif_data=True)
        generate_image(src_file, dst_file, settings)
        simple = get_exif_tags(get_exif_data(dst_file))
>       assert 'gps' in simple
E       AssertionError: assert 'gps' in {'Make': 'HTC-8900', 'Model': 'HTC-8900', 'dateobj': datetime.datetime(2008, 4, 6, 14, 55, 2), 'datetime': 'Sun 06 Apr 2008 02:55:02 PM '}

tests/test_image.py:255: AssertionError
----------------------------- Captured stderr call -----------------------------
The 'autorotate_images' and 'copy_exif_data' settings are not compatible because Sigal can't save the modified Orientation tag.
Failed to read GPS info
------------------------------ Captured log call -------------------------------
WARNING  sigal.image:image.py:98 The 'autorotate_images' and 'copy_exif_data' settings are not compatible because Sigal can't save the modified Orientation tag.
INFO     sigal.image:image.py:377 Failed to read GPS info
=========================== short test summary info ============================
FAILED tests/test_image.py::test_get_exif_tags - KeyError: 'fstop'
FAILED tests/test_image.py::test_exif_gps - AssertionError: assert 'gps' in {...
================= 2 failed, 69 passed, 2 deselected in 23.76s ==================
```

Full build log: 
[sigal-2.1-build.log](https://github.com/saimn/sigal/files/4901802/sigal-2.1-build.log)


I tried reverting to a pillow < 7.0.0 (we provide python-pillow6 for this), but python-pilkit can not import from it...).

I'm not entirely sure whether these issues are really caused by pillow >= 7.0.0 but it would at any rate be great to support it in the upcoming release to be on the safe side.

--- a/sigal/image.py
+++ b/sigal/image.py
@@ -48,6 +48,12 @@ from pilkit.utils import save_image
 from . import signals, utils
 from .settings import Status, get_thumb
 
+try:
+    # Pillow 7.2+
+    from PIL.TiffImagePlugin import IFDRational
+except ImportError:
+    IFDRational = None
+
 # Force loading of truncated files
 ImageFile.LOAD_TRUNCATED_IMAGES = True
 
@@ -304,9 +310,14 @@ def get_image_metadata(filename):
 def dms_to_degrees(v):
     """Convert degree/minute/second to decimal degrees."""
 
-    d = float(v[0][0]) / float(v[0][1])
-    m = float(v[1][0]) / float(v[1][1])
-    s = float(v[2][0]) / float(v[2][1])
+    if IFDRational and isinstance(v[0], IFDRational):
+        d = float(v[0])
+        m = float(v[1])
+        s = float(v[2])
+    else:
+        d = float(v[0][0]) / float(v[0][1])
+        m = float(v[1][0]) / float(v[1][1])
+        s = float(v[2][0]) / float(v[2][1])
     return d + (m / 60.0) + (s / 3600.0)
 
 
@@ -324,21 +335,29 @@ def get_exif_tags(data, datetime_format='%c'):
     if 'FNumber' in data:
         fnumber = data['FNumber']
         try:
-            simple['fstop'] = float(fnumber[0]) / fnumber[1]
+            if IFDRational and isinstance(fnumber, IFDRational):
+                simple['fstop'] = float(fnumber)
+            else:
+                simple['fstop'] = float(fnumber[0]) / fnumber[1]
         except Exception:
             logger.debug('Skipped invalid FNumber: %r', fnumber, exc_info=True)
 
     if 'FocalLength' in data:
         focal = data['FocalLength']
         try:
-            simple['focal'] = round(float(focal[0]) / focal[1])
+            if IFDRational and isinstance(focal, IFDRational):
+                simple['focal'] = round(float(focal))
+            else:
+                simple['focal'] = round(float(focal[0]) / focal[1])
         except Exception:
             logger.debug('Skipped invalid FocalLength: %r', focal,
                          exc_info=True)
 
     if 'ExposureTime' in data:
         exptime = data['ExposureTime']
-        if isinstance(exptime, tuple):
+        if IFDRational and isinstance(exptime, IFDRational):
+            simple['exposure'] = str(exptime)
+        elif isinstance(exptime, tuple):
             try:
                 simple['exposure'] = str(fractions.Fraction(exptime[0],
                                                             exptime[1]))

--- a/sigal/plugins/encrypt/endec.py
+++ b/sigal/plugins/encrypt/endec.py
@@ -1,5 +1,4 @@
 #!/usr/bin/env python3
-#coding: utf-8
 
 # copyright (c) 2020 Bowen Ding
 
@@ -34,7 +33,7 @@ from cryptography.hazmat.primitives.ciphers.aead import AESGCM
 from cryptography.hazmat.primitives.kdf.pbkdf2 import PBKDF2HMAC
 
 backend = default_backend()
-MAGIC_STRING = "_e_n_c_r_y_p_t_e_d_".encode("utf-8")
+MAGIC_STRING = b"_e_n_c_r_y_p_t_e_d_"
 
 def kdf_gen_key(password: str, salt: str, iters: int) -> bytes:
     password = password.encode("utf-8")

--- a/tests/test_image.py
+++ b/tests/test_image.py
@@ -150,7 +150,11 @@ def test_get_exif_tags():
     assert simple['iso'] == 50
     assert simple['Make'] == 'NIKON'
     assert simple['datetime'] == '22/01/2006'
-    assert simple['exposure'] == '100603/100000000'
+    try:
+        # Pillow 7.2+
+        assert simple['exposure'] == '0.00100603'
+    except Exception:
+        assert simple['exposure'] == '100603/100000000'
 
     data = {'FNumber': [1, 0], 'FocalLength': [1, 0], 'ExposureTime': 10}
     simple = get_exif_tags(data)

